src/build/app/settings.rs

```diff
@@ -689,6 +692,83 @@ pub enum AppSettings {
     /// [`subcommands`]: crate::App::subcommand()
     DeriveDisplayOrder,
 
+    /// Specifies that the parser should match first argument as [`subcommands`],
+    /// treat any matches as if the subcommand had been passed,
+    /// and print arguments like the [`subcommands`] are the top-level commands,
+    /// but if none match parse as usual.
+    ///
+    /// This is desired behaviour for "multicall" executables such as busybox
+    /// which dispatch to which applet to run
+    /// based on the name the executable has been linked as.
+    ///
+    /// # Panics
+    ///
+    /// If this option is combined with [`NoBinaryName`] then
+    /// [`try_get_matches_from_mut`] will [`panic!`].
+    ///
+    /// # Examples
+    ///
+    /// It does not make sense to combine NoBinaryName with Multicall
+    /// so it will panic.
+    ///
+    /// ```should_panic
+    /// # use clap::{App, AppSettings};
+    /// let _ = App::new("busybox")
+    ///     .setting(AppSettings::NoBinaryName)
+    ///     .setting(AppSettings::Multicall)
+    ///     .subcommand(App::new("true"))
+    ///     .subcommand(App::new("false"))
+    ///     .get_matches_from(&["true"]);
+    /// ```
+    ///
+    /// When the executable is run from a link named after an applet
+    /// that applet is matched.
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings};
+    /// let app = App::new("busybox")
+    ///     .setting(AppSettings::Multicall)
+    ///     .subcommand(App::new("true"))
+    ///     .subcommand(App::new("false"));
+    /// let m = app.get_matches_from(&["true"]);
+    /// assert_eq!(m.subcommand_name(), Some("true"));
+    /// ```
+    ///
+    /// When run from a link that is not named after an applet
+    /// the applet may be provided as a subcommand argument
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings};
+    /// # let app = App::new("busybox")
+    /// #     .setting(AppSettings::Multicall)
+    /// #     .subcommand(App::new("true"));
+    /// let m = app.get_matches_from(&["busybox", "true"]);
+    /// assert_eq!(m.subcommand_name(), Some("true"));
+    /// ```
+    ///
+    /// If the name of an applet is the name of the command,
+    /// the applet's name is matched
+    /// and subcommands may not be provided to select another applet.
```

Comment:
Should we combine these examples, with code comments highlighting different `get_matches_from` calls (to do so, we probably need `get_matches_from_mut`).  This might make it easier to highlight whats different between the two so people can see what this is doing faster.

Reply:
I didn't spot `get_matches_from_mut` when I wrote the original comment, thanks.

Reply:
Turns out there was a reason for that, it's `try_get_matches_from_mut`.

Reply:
Yeah, I can see us not providing every combination, thats a lot of functions.  You could
- Ignore my comment
- Use `clone` like the tesrs
- Use `try_get_matches_from_mut` and unwrap it or call `.exit()` on the error

Whats your thought?

Reply:
I think `try_get_matches_from_mut` then `unwrap` makes the most sense, since if `get_matches_from_mut` gets added it's easier to mechanically notice that it can be changed.